core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/DiagramServices.java

```diff
@@ -2100,13 +2102,20 @@ public void refreshRepresentationOfEditor(IEditorPart editor) {
    * @return
    */
   public EditPart getEditPart(DDiagramElement diagramElement) {
-    IEditorPart editor = EclipseUIUtil.getActiveEditor();
-    if (editor instanceof DiagramEditor) {
-      Session session = new EObjectQuery(diagramElement).getSession();
+    DDiagram parentDiagram = diagramElement == null ? null : diagramElement.getParentDiagram();
+    Session session = parentDiagram == null ? null : new EObjectQuery(parentDiagram).getSession();
+    IEditingSession uiSession = SessionUIManager.INSTANCE.getUISession(session);
+
```

Comment:
if uiSession is null, we can return null immediately

Reply:
Please consider the "return early" pattern.